### ISSUE TITLE
ReferenceError: requestAnimationFrame is not defined

### ISSUE THREAD

**Issue by freddiecabrera** [kirjs/react-highcharts#148]
Where exactly do you put 
```
if(!Highcharts){
  global.highcharts = require('react-highcharts/src/fakeHighcharts.js');
}
```
for node rendering? I have tried several different ways but I am not having any luck. Is there a complete example of this? Thanks in advanced.

**Comment by kirjs** [comment]
The example is a bit outdated, but should still work. 
You should put it at the point where highcharts has to be required. 
Better alternative would be to provide fake highcharts as highcharts dependency during in your build system. 

Regarding the error you're getting, try creating requestAnimationFrame in `fakeHighcharts.js`:
```js
var requestAnimationFrame = function(){};
```

If it works, please create a PR, or let me know, and I'll add it manually.

**Comment by freddiecabrera** [comment]
This is what my component currently looks like. I tried adding `var requestAnimationFrame = function(){};`
to `fakehighcharts.js` but I still got the error. Maybe I am doing it wrong. 
```

import React from 'react'
import Highcharts from 'highcharts'
if (!Highcharts) {
  global.highcharts = require('react-highcharts/src/fakeHighcharts.js')
}
import ReactHighcharts from 'react-highcharts'
import { performanceScoreByMatch, performanceScoreCumulative } from '../../temp/guardado'

const date = performanceScoreByMatch.attacking.map(data => data.date)

const performanceScoreByMatchAttackingScore = performanceScoreCumulative.attacking.map(data => JSON.parse(data.score))
const performanceScoreByMatchDefensiveScore = performanceScoreCumulative.defensive.map(data => JSON.parse(data.score))
const performanceScoreByMatchOverallScore = performanceScoreCumulative.overall.map(data => JSON.parse(data.score))
const performanceScoreByMatchPossesionScore = performanceScoreCumulative.possesion.map(data => JSON.parse(data.score))

// const performanceScoreCumulativeAttackingScore = performanceScoreCumulative.attacking.map(data => data.score)
// const performanceScoreCumulativeDefensiveScore = performanceScoreCumulative.defensive.map(data => data.score)
// const performanceScoreCumulativeOverallScore = performanceScoreCumulative.overall.map(data => data.score)
// const performanceScoreCumulativePossesionScore = performanceScoreCumulative.possesion.map(data => data.score)

const config = {
  chart: {
    backgroundColor: '#0e0e13',
    type: 'areaspline'
  },
  title: {
    text: 'Performance Score By Match'
  },
  xAxis: {
    categories: date
  },
  plotOptions: {
         areaspline: {
             fillOpacity: 0.1
         }
     },
  series: [
    { name: 'Attacking', data: performanceScoreByMatchAttackingScore, color: 'green' },
    { name: 'Defensive', data: performanceScoreByMatchDefensiveScore, color: 'red' },
    { name: 'Possesion', data: performanceScoreByMatchPossesionScore, color: 'yellow' },
    { name: 'Overall', data: performanceScoreByMatchOverallScore, color: 'blue' }
  ]
}

const Chart = React.createClass({
  render () {
    return (
      <div>
        <ReactHighcharts config={config} />
      </div>
    )
  }
})

export default Chart
```

**Comment by kirjs** [comment]
how about `global.requestAnimationFrame = function(){};`?

**Comment by freddiecabrera** [comment]
Nothing seems to be making it work. The chart gets rendered but it gets that error 

`bundle.js:33065 Uncaught TypeError: Cannot read property 'requestAnimationFrame' of undefinedrenderChart @ bundle.js:33065HighchartsChart_componentDidMount @ bundle.js:33073notifyAll @ bundle.js:6436close @ bundle.js:16224closeAll @ bundle.js:6797perform @ bundle.js:6744batchedMountComponentIntoNode @ bundle.js:2637perform @ bundle.js:6731batchedUpdates @ bundle.js:10781batchedUpdates @ bundle.js:6236_renderNewRootComponent @ bundle.js:2831ReactMount__renderNewRootComponent @ bundle.js:1425_renderSubtreeIntoContainer @ bundle.js:2905render @ bundle.js:2925React_render @ bundle.js:1425(anonymous function) @ bundle.js:63__webpack_require__ @ bundle.js:20(anonymous function) @ bundle.js:40(anonymous function) @ bundle.js:43`

**Comment by freddiecabrera** [comment]
Here is the Repo if you would like to check it out https://github.com/freddiecabrera/ullibol-client

**Comment by freddiecabrera** [comment]
I got this to work by removing any mention about requestAnimationFrame. I am doing server rendering and I did not have to do
```
 if (!Highcharts) {
  global.highcharts = require('react-highcharts/src/fakeHighcharts.js')
}
```
I just did `import ReactHighcharts from 'react-highcharts'`. Is there a way I can get a version without requestAnimationFrame?

**Comment by rxt2012kc** [comment]
I encounter the same problem as ahead, when I require('react-highcharts/dist/ReactHighcharts'), the console shows ReferenceError: requestAnimationFrame is not defined

**Comment by kirjs** [comment]
Let me know if you still need help with it, I'll ping you on IM and we'll figure it out.

**Comment by freddiecabrera** [comment]
I just completely removed anything having to do with requestAnimationFrame and I got it to work.

**Comment by kirjs** [comment]
Which version of node did you use?

**Comment by jontore** [comment]
I've also encountered this problem when I package my application, and react-highcharts is packaged with webpack.

**Comment by chevin99** [comment]
Same problem for me as @jontore:

> I've also encountered this problem when I package my application, and react-highcharts is packaged with webpack.

**Comment by freddiecabrera** [comment]
I was using node 4.3.0

@jontore @chevin99 if you guys would like to get passed it just go through all the files and manually remove anything having to do with requestAnimationFrame. I am sure that is not the best way to do it but it worked for me.

**Comment by kirjs** [comment]
Can somebody help me reproduce that?

**Comment by chevin99** [comment]
@kirjs here you go:

app.js -

```
let React = require('react')
let ReactDOM = require('react-dom')
let ReactHighcharts = require('react-highcharts')

let Chart = React.createClass({
  render: function () {
    let config = {
      series: [{
        name: 'A',
        data: [1, 2, 3, 4]
      }, {
        name: 'B',
        data: [5, 4, 3, 2]
      }]
    }

    return <ReactHighcharts config={config}/>
  }
})

ReactDOM.render(<Chart/>, document.getElementById('app'))
```

In my webpack config I use the babel loader:

```
module: {
    loaders: [
      {
        test: /\.js?$/,
        loader: 'babel',
        query: {
          presets: ['es2015', 'react', 'stage-0']
        }
      }
    ]
  }
```

Hope that helps.  Thanks.

**Comment by kirjs** [comment]
I tried to reproduce again, but couldn't. 
Which browser/environment? which react-highcharts version? What's the exact error message?

**Comment by chevin99** [comment]
Browser: Chrome Version 51.0.2704.103 (64-bit)
react-highcharts version 8.3.3

There error is just `Uncaught TypeError: Cannot read property 'requestAnimationFrame' of undefined`

I believe the problem is line 33 in **chartsFactory.jsx**:

`global.requestAnimationFrame && requestAnimationFrame(()=>{`

`global` isn't defined in the browser.  If I update the code to have `global` default to `window` the error goes away.

**Comment by kirjs** [comment]
Should be fixed in 8.4.1, feel free to reopen if not
